### ISSUE TITLE
Segfault when exporting ONNX graph from transformers library

### ISSUE THREAD

**Issue by Narsil** [pytorch/pytorch#53383]
## 🐛 Bug

When trying to export an ONNX graph, there is a segfault

## To Reproduce

1. `git checkout git@github.com:huggingface/transformers.git`
2. `git checkout 395ffcd757103ed2ccc888e48d90fd2ccb4d506f` (master has the test skipped currently)
3. `pytest tests/test_modeling_t5.py::T5ModelTest::test_export_to_onnx`

Edit:

Forgot to include the actual code just to see it more directly:

```python
def test_export_to_onnx(self):                                             
    config_and_inputs = self.model_tester.prepare_config_and_inputs()      
    model = T5Model(config_and_inputs[0]).to(torch_device)                 
    with tempfile.TemporaryDirectory() as tmpdirname:                      
        torch.onnx.export(                                                 
            model,   
           # Input ids: (13 batch size, 7 seq_length) LongTensor
           # Decoder input ids: (13 batch size, 9 seq_length) LongTensor
           # Attention mask, (13, 7) LongTensor                                         
            (config_and_inputs[1], config_and_inputs[3], config_and_inputs[2]),
            f"{tmpdirname}/t5_test.onnx",                                  
            export_params=True,                                            
            opset_version=9,                                               
            input_names=["input_ids", "decoder_input_ids"],    
```

## Expected behavior

At the very least a clean error.
Ideally it should work like it used to.

This test is a simple export of T5 architecture using (`input_ids`, `attention_mask` and `decoder_input_ids`)

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

PyTorch version: 1.8.0   *pip install -U torch
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: 10.0.0-4ubuntu1 
CMake version: version 3.16.3

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect  (It's running 11.2 but the bug is CPU anyway)
GPU models and configuration: GPU 0: GeForce GTX 970
Nvidia driver version: 460.32.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] torch==1.8.0
[conda] Could not collect  # Using pyenv

## Additional context

Tried to run GDB, and figured the problem lied there:
```
hread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007fffc8c61fd3 in at::Tensor::is_mkldnn() const ()
   from /home/nicolas/src/transformers/.venv/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so
(gdb) bt
#0  0x00007fffc8c61fd3 in at::Tensor::is_mkldnn() const ()
   from /home/nicolas/src/transformers/.venv/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so
#1  0x00007fffc81d4678 in c10::TensorType::create(at::Tensor const&) ()
   from /home/nicolas/src/transformers/.venv/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so
#2  0x00007fffda0f9642 in torch::jit::ONNXAssignOutputShape(std::shared_ptr<torch::jit::Graph>&, c10::ArrayRef<at::Tensor>, torch::jit::python::IODescriptor const&, bool) () from /home/nicolas/src/transformers/.venv/lib/python3.8/site-packages/torch/lib/libtorch_python.so
#3  0x00007fffda071207 in pybind11::cpp_function::initialize<torch::jit::initJITBindings(_object*)::{lambda(std::shared_ptr<torch::jit::Graph>&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, torch::jit::python::IODescriptor const&, bool)#1}, void, std::shared_ptr<torch::jit::Graph>&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, torch::jit::python::IODescriptor const&, bool, pybind11::name, pybind11::scope, pybind11::sibling>(torch::jit::initJITBindings(_object*)::{lambda(std::shared_ptr<torch::jit::Graph>&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, torch::jit::python::IODescriptor const&, bool)#1}&&, void (*)(std::shared_ptr<torch::jit::Graph>&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, torch::jit::python::IODescriptor const&, bool), pybind11::name const&, pybind11::scope const&, pybind11::sibling const&)::{lambda(pybind11::detail::function_call&)#3}::_FUN(pybind11::detail::function_call) ()
   from /home/nicolas/src/transformers/.venv/lib/python3.8/site-packages/torch/lib/libtorch_python.so
#4  0x00007fffd9cab81e in pybind11::cpp_function::dispatcher(_object*, _object*, _object*) ()
   from /home/nicolas/src/transformers/.venv/lib/python3.8/site-packages/torch/lib/libtorch_python.so
#5  0x00005555555ce52a in cfunction_call_varargs (kwargs=0x7fff8cf6edb0, args=0x7fffef43bb30, func=0x7fff8cf70f40) at Objects/call.c:742
#6  PyCFunction_Call (func=func@entry=0x7fff8cf70f40, args=args@entry=0x7fffef43bb30, kwargs=kwargs@entry=0x0) at Objects/call.c:772
#7  0x00005555555cad01 in _PyObject_MakeTpCall (callable=callable@entry=0x7fff8cf70f40, args=args@entry=0x55555bd311a8, 
    nargs=<optimized out>, keywords=keywords@entry=0x0) at Objects/call.c:159
#8  0x00005555555bb366 in _PyObject_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, args=<optimized out>, 
    callable=<optimized out>) at ./Include/cpython/abstract.h:125
#9  call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555b1ce30) at Python/ceval.c:4963
```

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @SplitInfinity

**Comment by Narsil** [comment]
Hi any news on this? 

Willing to help if needed.

**Comment by Narsil** [comment]
Seems fixed on 1.8.1
